internal/agent/agent.go

```
@@ -140,8 +148,18 @@ func (a *Agent) startMetricsServer(ctx context.Context) error {
 	mux.HandleFunc("/livez", a.handleLiveness)
 	mux.HandleFunc("/readyz", a.handleReadiness)
 	
+	addr := ":8080"
+	if a.config != nil && a.config.MetricsAddr != "" {
+		addr = a.config.MetricsAddr
+	}
+
+	ln, err := net.Listen("tcp", addr)
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)